torchio/data/queue.py

```diff
@@ -250,14 +296,24 @@ def _get_first_item(batch):
     def _get_subjects_iterable(self) -> Iterator:
         # I need a DataLoader to handle parallelism
         # But this loader is always expected to yield single subject samples
+
+        # Same random shuffling applied to subjects and volumes
+        if self.shuffle_subjects:
+            random_idx = list(RandomSampler(self.subjects_dataset))
```

Comment:
Have you tried this? I might be wrong, but I suspect this will load all the data in the dataset from disk, which is really undesirable.

Reply:
I haven't tested RandomSampler with real data, but by looking at the [https://github.com/pytorch/pytorch/blob/master/torch/utils/data/sampler.py#L113](code) it seems that it doesn't load the data; it works directly with len(data). In the end, random_idx becomes a randomized list of indices